Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyBoolComboBoxCtrl.hxx

```diff
@@ -24,7 +25,7 @@ class QPushButton;
 
 namespace AzToolsFramework
 {
-    //just a test to see how it would work to pop a dialog
+    using ComboBoxSelectFunc = AZ::Edit::AttributeFunction<bool(bool)>;
```

Comment:
We shouldn't be caching attributes locally.
Instead the value contained within the Attribute can be read from it.
Furthermore what should be read is an `AZStd::function<bool(bool>` from the Attribute as that would allow functors and lambdas to be used for the combo box select function.

Reply:
Also API documentation as to what value the user can except to be passed into the function and the what the user should return needs to be added.
Such as
```c++
//! Type Alias for the Error String type that is used when an Edit is rejected
//! Uses AZStd::fixed_string to avoid dynamic memory allocations
using EditRejectString = AZStd::fixed_string<256>;

//! Outcome that returns the value that the ComboBox control should set on Success
//! On Failure a rejection string with the reason for failure is returned
using EditResultOutcome = AZ::Outcome<bool, EditRejectString>;

//! Callback to implement that determines the new value to update the Combo Box Ctrl with
//! or to provide the Combo Box Ctrl system with an error message as to why the new value could not be set
//! i.e "new value not convertible to type", "new value is out of range", etc...
//! @param currentValue The current value stored in the combo box
//! @return Outcome which should store the new value to set in the combo box control or a string supplying the reason why the edit has been rejected.
using EditButtonCallback = AZStd::function<EditResultOutcome(bool currentValue)>;
```